src/uct/ib/rc/base/rc_iface.c

```diff
@@ -556,6 +556,68 @@ uct_rc_iface_init_max_rd_atomic(uct_rc_iface_t *iface,
     return UCS_OK;
 }
 
+static ucs_status_t
+uct_rc_iface_recv_sg_mpools_init(uct_ib_iface_t *iface,
+                                 const uct_ib_iface_config_t *config,
+                                 const uct_iface_params_t *params,
+                                 const char *name, ucs_mpool_t *mp)
+{
+    size_t align_offset, alignment;
+    ucs_status_t status;
+    unsigned grow;
+
+    if (config->rx.queue_len < 1024) {
+        grow = 1024;
+    } else {
+        /* We want to have some free (+10%) elements to avoid mem pool expansion */
+        grow = ucs_min((int)(1.1 * config->rx.queue_len + 0.5),
+                       config->rx.mp.max_bufs);
+    }
+
+    /* Preserve the default alignment by UCT header if user does not request
+     * specific alignment.
+     * TODO: Analyze how to keep UCT header aligned by cache line even when
+     * user requested specific alignment for payload.
+     */
+    status = uct_iface_param_am_alignment(params,
+                                          iface->config.rx_payload_offset, 0,
+                                          iface->config.rx_hdr_offset,
+                                          &alignment, &align_offset);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    status = uct_iface_mpool_init(&iface->super,
+                                  &mp[UCT_IB_RX_SG_TL_HEADER_IDX],
+                                  iface->config.rx_payload_offset, align_offset,
+                                  alignment, &config->rx.mp, grow,
+                                  uct_ib_iface_recv_desc_init, name);
+
+    status = uct_iface_param_am_alignment(params,
```

Comment:
probably could remove am_alignment since we now require user allocator
also, need only pool for descs+headers